caip-300.md

```diff
@@ -0,0 +1,223 @@
+  - code = 7001
+  - message = "Invalid Method Requested"
+
+### Fallback Behavior
+
+For backwards-compatibility, there is going to be a fallback behavior that is expected but is going to dependant on each ecosystem that must be defined under the [Namespaces][namespaces] repo to define a CAIP-300 profile.
+
+If we take for example Ethereum this would fallback to the legacy behavior of `eth_requestAccounts` as defined by ERC-1102
+
+```jsonc
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "wallet_connect",
+  "params": {} // fallback to ERC-1102 request
+}
+```
+
+Which then will result in a single item in the array that will match the response of the fallback method defined for each Namespace profile for CAIP-300.
+
+```jsonc
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": [
+    {
+      "result": ["0xa...", "0xb..."] // fallback to ERC-1102 request
+    }
+  ]
+}
+```
+
+## Rationale
+
+An RPC method was choosen for batching because it enabled a smoother upgrade path for existing dapps and wallets to support it without much breaking changes. While the alternative would be to batch RPC requests at the transport layer this is not possible today without a breaking change.
+
+Therefore batching inside a single RPC request makes it easier to re-use the existing developer tooling and will enable both dapps and wallet to progressively support this new "connection" standard for wallets.
+
+The rationale for making this as generic as possible is to enable profiles to be defined for different ecosystems such as Bitcoin, Ethereum, Solana, Cosmos, Polkadot, etc. The alternative would be to create specific fields for "session", "authentication", "capabilities", "permissions", etc. But this would introduce a bias from one chain to another and not favor chain-agnosticisim.
+
+Finally it was intentional that batch requests do not have `id` in the schema because all requests are intended to be resolved together in the same round-trip and they must be ordered as requested.
+
+## Test Cases
+
+Let's take for example the Ethereum ecosystem which currently has popularized the session creation with CAIP-25 being followed by an authentication request with CAIP-222. Yet there are new standards being developed where other functionality is defined pre-session and requires being batched together. This example will include capability expsosure with ERC-5792 and permission granting with ERC-7715.
+
+```jsonc
+{
+  "id": 123,
+  "jsonrpc": "2.0",
+  "method": "wallet_connect",
+  "params": {
+    "requests?": [
+      {
+          "method": "wallet_authenticate",  // CAIP-222 request
+          "params": {...}
+      },
+      {
+          "method": "wallet_createSession", // CAIP-25 request
+          "params": {...}
+      },
+      {
+          "method": "wallet_getCapabilities", // ERC-5792 request
+          "params": {...}
+      },
+      {
+          "method": "wallet_grantPermissions", // ERC-7715 request
+          "params": {...}
+      },
```

Comment:
Does this standard allows for `personal_sign` to be sent via `wallet_connect`? As at the time of sending, the user address is unknown to the dapp

Reply:
Not really... you need to use wallet_authenticate